### PR TITLE
Adds template rendering module and server-side mustache

### LINKED ISSUE #7: Mustache rendering module
This endpoint is similar to XHR proxy, but also performs server-side rendering of Mustache.js templates.

After a request is completed following the same steps the XHR proxy does, this module does the following:

-   Uses the JSON output to render the given mustache template via mustache.js
-   Run the logic provided by [purifier.js](https://github.com/ampproject/amphtml/blob/master/src/purifier.js) to ensure the template output is sanitized
-   Rewrite all image URLs to use image proxy
-   Rewrite all hyperlinks to use redirect endpoint and include `target=_blank` and `rel=noreferrer noopener`

### PR DESCRIPTION
**Pull request by fstanis**
Closes #7 and #11

### CODE REVIEW

**Comment by alabiaga** [comment]
sorry for delay. Will look at this today. Thanks

File: example/backend/request.js [hunk 1]
```diff
@@ -95,9 +95,6 @@ function validateRequest(request) {
 
 function validateResponse(response) {
   const { headers, status } = response;
-  if (status !== 200) {
-    return false;
-  }
   if (!isJSON(headers.get('content-type'))) {
```
Comment (alabiaga) [inline example/backend/request.js]: I find it odd that we are passing in the content type string to verify that response is of JSON type. We should pass in the entire response and do the JSON verifying in the method body.

File: src/config.ts [hunk 2]
```diff
@@ -22,9 +22,24 @@ export interface Config {
...
+  /**
+   * URL of the template proxy. If unset, template rendering is not proxied.
+   */
+  templateProxyURL?: string;
+
+  /**
+   * Runs some transforming preprocessing modules on the output received from
+   * the template proxy.
+   *
+   * Must be false if templateProxyURL is unset.
```
Comment (alabiaga) [inline src/config.ts]: do we enforce this check in the code somewhere?

File: example/backend/request.js (hunk 1 above)
Comment (fstanis) [reply example/backend/request.js]: So, we call [.json()](https://developer.mozilla.org/en-US/docs/Web/API/Body/json) which throws on invalid JSON from `requestWithCORS`. This is just to halt any further attempt to deal with the response if the `content-type` is invalid.

File: src/config.ts (hunk 2 above)
Comment (fstanis) [reply src/config.ts]: Hm, good point: `validateConfig` does check this, but it seems we're not calling it anywhere right now.

File: example/backend/request.js (hunk 1 above)
Comment (alabiaga) [reply example/backend/request.js]: I see. Can we rename method. IsJsonContentTypeHeader?

**Pull request merged**